crates/uv-requirements/src/lookahead.rs

```diff
@@ -52,18 +54,34 @@ impl<'a> LookaheadResolver<'a> {
         context: &T,
         client: &RegistryClient,
     ) -> Result<Vec<RequestedRequirements>> {
-        let requirements: Vec<_> = futures::stream::iter(self.requirements.iter())
-            .map(|requirement| async { self.lookahead(requirement, context, client).await })
-            .buffered(50)
-            .try_collect()
-            .await?;
-        Ok(requirements.into_iter().flatten().collect())
+        // TODO(charlie): Apply overrides and constraints.
```

Comment:
Uncertain how to handle these... I don't think we want to blindly resolve them recursively, because we can't _know_ that a given constraint or override will _actually_ be applied, and so we might end up incorporating requirements that won't end up being correct here.

E.g., consider package `foo @ bar`, which has a URL requirement of `baz @ bop`. Assume `foo @ bar` is a constraint. If we include it here, then we'll also include `baz @ bop` as a required URL. But what if no one ever requests `foo`, and so the constraint should have no effect? Now we've added a `baz @ bop` requirement that shouldn't be enforced...

I think the best we can do (under this paradigm) is enforce overrides and constraints for the dependencies we see during _this_ resolution pass.